core/tool_cache.py

```
@@ -0,0 +1,108 @@
+        key = f"tc_{len(self._cache)}_{int(time.time() * 1000)}"
+        self._cache[key] = result
+        if len(self._cache) > self._max_size:
+            self._cache.popitem(last=False)  # 淘汰最旧
+        return key
+
+    def retrieve(self, key: str) -> Optional[Dict[str, Any]]:
+        """取出完整结果，不存在返回 None。"""
+        return self._cache.get(key)
+
+    def summarize(self, tool_name: str, result: Dict[str, Any]) -> Dict[str, Any]:
+        """将完整结果转换为摘要，不含 matches/content 等大字段。"""
```

Ruff (0.15.10)

[warning] Docstring contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF002)